packages/azure-http-specs/specs/azure/resource-manager/method-subscription-id/main.tsp

```diff
@@ -0,0 +1,478 @@
+      SubscriptionResource2,
+      BaseParameters = GetSubscriptionResource2BaseParameter
+    >;
+
+    @scenario
+    @scenarioDoc("""
+      Resource PUT operation for SubscriptionResource2 with method-level subscriptionId.
+      Expected path: /subscriptions/00000000-0000-0000-0000-000000000000/providers/Azure.ResourceManager.MethodSubscriptionId/subscriptionResource2s/sub-resource-2
+      Expected query parameter: api-version=2023-12-01-preview
+      Expected request body:
+      ```json
+      {
+        "properties":{
+          "configValue": "test-config"
+        }
+      }
+      ```
+      Expected response body:
+      ```json
+      {
+        "id": "/subscriptions/00000000-0000-0000-0000-000000000000/providers/Azure.ResourceManager.MethodSubscriptionId/subscriptionResource2s/sub-resource-2",
+        "name": "sub-resource-2",
+        "type": "Azure.ResourceManager.MethodSubscriptionId/subscriptionResource2s",
+        "properties":{
+          "configValue": "test-config",
+          "provisioningState": "Succeeded"
+        },
+        "systemData": {
+          "createdBy": "AzureSDK",
+          "createdByType": "User",
+          "createdAt": "2023-01-01T00:00:00.000Z",
+          "lastModifiedBy": "AzureSDK",
+          "lastModifiedAt": "2023-01-01T00:00:00.000Z",
+          "lastModifiedByType": "User"
+        }
+      }
+      ```
+      """)
+    put is ArmResourceCreateOrReplaceSync<
+      SubscriptionResource2,
+      BaseParameters = PutSubscriptionResource2BaseParameter
+    >;
+
+    @scenario
+    @scenarioDoc("""
+      Resource DELETE operation for SubscriptionResource2 with method-level subscriptionId.
+      Expected path: /subscriptions/00000000-0000-0000-0000-000000000000/providers/Azure.ResourceManager.MethodSubscriptionId/subscriptionResource2s/sub-resource-2
+      Expected query parameter: api-version=2023-12-01-preview
+      Expected response status code: 204
+      """)
+    delete is ArmResourceDeleteSync<
+      SubscriptionResource2,
+      BaseParameters = DeleteSubscriptionResource2BaseParameter
+    >;
+  }
+}
+
+/**
+ * Scenario 2: One subscription level resource (method-level subscriptionId) and one resource group level resource (client-level subscriptionId)
+ *
+ * Test mixed parameter placement: subscription resource with method-level subscriptionId and resource group resource with client-level subscriptionId.
+ *
+ * This scenario has:
+ * 1. One subscription-level resource (SubscriptionResource) with subscriptionId moved to method level
+ * 2. One resource group-level resource (ResourceGroupResource) with subscriptionId staying at client level
+ *
+ * Expected behavior:
+ * - Client should have subscriptionId parameter in initialization (for ResourceGroupResource operations)
+ * - SubscriptionResource operations should have subscriptionId as method-level parameter
+ * - ResourceGroupResource operations should not have subscriptionId as method-level parameter (uses client-level)
+ */
```

Comment:
Could we add following two operations?
- without subcription id and operation list: https://github.com/marygao/azure-rest-api-specs/blob/relay-tsp-migraiton/specification/resourcehealth/resource-manager/Microsoft.ResourceHealth/stable/2025-05-01/ResourceHealth.json#L127
- without subscript id but not operation list https://github.com/marygao/azure-rest-api-specs/blob/relay-tsp-migraiton/specification/resourcehealth/resource-manager/Microsoft.ResourceHealth/stable/2025-05-01/ResourceHealth.json#L257

Reply:
Another question is - if we have some tenant-level operations but customers forget to de-prompt the subscription id parameter, what values to input in constructor?

Reply:
It will in client level. I don't think it is an easy lint. But you could open a follow up issue.

Reply:
Offline discussed, i don't think we need a linter. @tadelesh could you add two tenant-level operations for js testing purpose? Thanks!